libandroid/app/src/main/java/com/mapbox/services/android/testapp/distance/DistanceActivity.java

```diff
@@ -0,0 +1,340 @@
+package com.mapbox.services.android.testapp.distance;
+
+import android.graphics.Color;
+import android.graphics.PointF;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+
+import com.google.gson.JsonObject;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.style.layers.CircleLayer;
+import com.mapbox.mapboxsdk.style.layers.NoSuchLayerException;
+import com.mapbox.mapboxsdk.style.layers.Property;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+import com.mapbox.mapboxsdk.style.sources.Source;
+import com.mapbox.services.android.testapp.R;
+import com.mapbox.services.android.testapp.Utils;
+import com.mapbox.services.commons.ServicesException;
+import com.mapbox.services.commons.geojson.Feature;
+import com.mapbox.services.commons.geojson.FeatureCollection;
+import com.mapbox.services.commons.geojson.Point;
+import com.mapbox.services.commons.models.Position;
+import com.mapbox.services.directions.v5.DirectionsCriteria;
+import com.mapbox.services.distance.v1.MapboxDistance;
+import com.mapbox.services.distance.v1.models.DistanceResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class DistanceActivity extends AppCompatActivity implements OnMapReadyCallback, MapboxMap.OnMapClickListener {
+
+  private static final String TAG = "DistanceActivity";
+  private static final String KEY_DURATION = "duration";
+  private static final String KEY_NAME = "name";
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private List<Position> coordinates;
+  private List<Feature> restaurants;
+
+  private List<Feature> features;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_distance);
+
+    Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+    coordinates = new ArrayList<>();
```

Comment:
Initialisation can go into the field declaration as it's only done once anyway. That way you can make them final and there's no chance of null pointer exceptions (when refactoring)